src/polynomial.h

```diff
@@ -176,23 +176,19 @@ void Polynomial<T>::set(int exponent, T coef)
     assert(exponent >= 0);
     assert(field->check(coef));
 
-    typename Term<T>::const_iterator it = terms.find(exponent);
-
+    auto it = terms.find(exponent);
     if (it == terms.end()) {
         if (coef == 0)
             return;
     } else {
-        if (coef == 0) {
+        if (coef == 0 && exponent == _degree) {
```

Comment:
Why these changes? You could add some explanation in the commit message?

It super useful when you git blame some code to have some context/explanation in the commit message.